packages/blocks/src/lib/maplibre/hooks.ts

```diff
@@ -0,0 +1,49 @@
+import { getContext, onDestroy } from 'svelte';
+import type {
+  LngLat,
+  Map,
+  MapLayerEventType,
+  MapLayerMouseEvent,
+  MapLayerTouchEvent,
+} from 'maplibre-gl';
+import { type Writable, get } from 'svelte/store';
+
+/**
+ * Provides context for a <MapLibre> instance. Must be called within a child of this component.
+ */
+export const useMapLibre = () => {
+  const mapContext = getContext<Writable<Map> | undefined>('map');
+
+  if (!mapContext) {
+    throw new Error(
+      'useMapLibre is a context sensitive hook that must be used inside a <MapLibre> component.'
+    );
+  }
+
+  const map = get(mapContext);
+  const mapCenter = getContext<Writable<LngLat>>('center');
+  const mapSize =
+    getContext<Writable<{ width: number; height: number }>>('size');
+  const mapZoom = getContext<number>('zoom');
```

Comment:
Can we combine these three contexts (map, size, and zoom) into a single context with multiple stores?

Reply:
yep, good idea 👍